tests/calico_cni_k8s_test.go

```diff
@@ -1669,6 +1781,9 @@ var _ = Describe("Kubernetes CNI tests", func() {
 			// Delete the container.
 			_, err = testutils.DeleteContainer(netconfCalicoIPAM, netNS.Path(), name, testutils.K8S_TEST_NS)
 			Expect(err).ShouldNot(HaveOccurred())
+
+			// Delete pod
+			ensurePodDeleted(clientset, testutils.K8S_TEST_NS, name)
```

Comment:
Can we do this in an afterEach as well?

Reply:
I'm still a bit curious as to why these passed before, but now need this.. feel like I must be missing something!

Reply:
> Can we do this in an afterEach as well?

Yes. I have updated the code so that, if there is only one test case, I would use 
```defer ensurePodDeleted(clientset, testutils.K8S_TEST_NS, name)``` right after pod been created. 

Otherwise, `ensurePodDeleted` is calling from `afterEach` for multiple test cases.

Reply:
> I'm still a bit curious as to why these passed before, but now need this.. feel like I must be missing something!

Test passed before with etcd datastore. This is because `wipeDatastore` cleanup calico resources include WEPs. In kubernetes datastore, we need to clean up Pod as WEP resources for each test.

Reply:
> defer ensurePodDeleted(clientset, testutils.K8S_TEST_NS, name) right after pod been created.

My preference is to simply use BeforeEach and AfterEach anyway, since we might add additional tests to those contexts later and the mix of defers / AfterEach adds cognitive overhead. Let's not worry about adjusting that now, but we can clean it up in the future.